comp/core/tagger/impl-remote/remote.go

```diff
@@ -354,10 +354,9 @@ func (t *remoteTagger) DogstatsdCardinality() types.TagCardinality {
 }
 
 // Subscribe returns a channel that receives a slice of events whenever an entity is
-// added, modified or deleted. It can send an initial burst of events only to the new
-// subscriber, without notifying all of the others.
-func (t *remoteTagger) Subscribe(subscriptionID string, filter *types.Filter) (types.Subscription, error) {
-	return t.store.subscribe(subscriptionID, filter)
+// Agents running the remote tagger don't have the ability to subscribe to events, as they are the ones receiving the events.
```

Comment:
I am not sure I fully understand this.

An agent using the remote tagger will be receiving tags from the server, this is correct. However, why wouldn't it be possible for another component in the same agent to subscribe to the remote tagger?

A valid use case might be the following:

Suppose we have a CLC runner that is running the remote tagger, receiving tags from the DCA server. What if we want to add a component/feature in the CLC runner which subscribers to the remote tagger to get pod tag updates for some reason, this should be possible by calling `remoteTagger.Subscribe`.

I understand that currently we don't have such a use case, but it is at least possible to have easily in the future. 

If there is a strong need to reduce the size of the remote tagger by removing subscription capability, I think the Subscribe method of the remote tagger should not be returning a `nil` error, but rather something like:

```
return nil, errors.New("subscription to the remote tagger is not current supported")
```

Reply:
> Suppose we have a CLC runner that is running the remote tagger, receiving tags from the DCA server. What if we want to add a component/feature in the CLC runner which subscribers to the remote tagger to get pod tag updates for some reason, this should be possible by calling remoteTagger.Subscribe.

> I understand that currently we don't have such a use case, but it is at least possible to have easily in the future.

I don't disagree with anything you said there, as for potential use cases. 

I disagree with the idea of having functionality in the Agent that is not used with the premiss that it might be used in the future. Programming thinking into the future might lead to early optimization or bloated codebase. The Agent is used by a lot of customers, we should strive to have the less amount of unused code/functionality to avoid any extra overhead. 

> I think the Subscribe method of the remote tagger should not be returning a nil error, but rather something like

I'm ok modifying the returned error to provide more information. Thanks for the suggestion 👍

Reply:
Thanks 😄